### PR TITLE
Custom matchers feature

### PR DESCRIPTION
**Pull request by bandersongit**
Add support for custom matchers. Depends on https://github.com/facebookexperimental/reason-native/pull/63, targeting that branch for readability of the PR. See the CustomMatchers_test.re file for usage

### CODE REVIEW

File: src/rely/Rely.re [hunk 1]
```diff
@@ -354,31 +386,36 @@ module Make = (UserConfig: FrameworkConfig) => {
...
-                 r.onTestSuiteResult(
-                   reporterSuite,
-                   newResult,
-                   testSuiteResult,
-                 )
-               );
-               {...acc, aggregatedResult: newResult};
+               switch(testSuite){
+               | TestSuiteInternal({name, fn, matchersExtensionFn}) => {
+                let reporterSuite = {name: name};
```
[the remark this answers is not in the document]
Comment (bandersongit) [reply src/rely/Rely.re]: With a single field the compiler doesn't actually figure out that this is a record. In this case ```let reporterSuite = {name}``` results in reporterSuite being assigned the string value that is name, if there was another field punning would work.

File: src/rely/matchers/MatcherUtils.rei [hunk 2]
```diff
@@ -24,4 +24,4 @@ type matcherUtils = {
   indent: string => string,
 };
 
-let matcherUtils: matcherUtils;
+let matcherUtils: t;
```
[the remark this answers is not in the document]
Comment (bandersongit) [reply src/rely/matchers/MatcherUtils.rei]: It wasn't exposed publicly previously and I updated references.

File: src/refmterr/lib/dune [hunk 3]
```diff
@@ -13,3 +13,4 @@
 (library (name Refmterr)
     (public_name refmterr.lib)
     (libraries atdgen re.pcre unix pastel.lib))
+(copy_files# ../../../common/*)
```
Comment (jordwalke) [inline src/refmterr/lib/dune]: Is this related to this diff?

**Comment by jordwalke** [comment]
I don't think this diff is reviewable because it doesn't appear to actually be "on top of" the previous commit: https://github.com/facebookexperimental/reason-native/pull/63/files

I noticed the `esy.lock/index.json` is changed in both commits, and both add a `Reporter.re` file.

File: src/refmterr/lib/dune (hunk 3 above)
Comment (bandersongit) [reply src/refmterr/lib/dune]: This was the same error you fixed separately. I thought it came out of a merge conflict.

**Comment by bandersongit** [comment]
> I don't think this diff is reviewable because it doesn't appear to actually be "on top of" the previous commit: https://github.com/facebookexperimental/reason-native/pull/63/files
> 
> I noticed the `esy.lock/index.json` is changed in both commits, and both add a `Reporter.re` file.

Yeah it looks like it got pretty messed up. I'm landing the reporters work now, once shipit gets that code here then I'll rebase this and clean things up.
